### ISSUE TITLE
'maestro' is only available in iOS 12.0 or newer

### ISSUE THREAD

**Issue by Shmublon** [stripe/stripe-react-native#970]
**Describe the bug**
When I trying to build ios version app I have the error "'maestro' is only available in iOS 12.0 or newer", but my 
target ios version is 12.0

**To Reproduce**
1. Install stripe
2. Run build in Xcode

**Expected behavior**
Build success

React native info: 
System:
    OS: macOS 11.6.5
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 926.85 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 8.1.4 - ~/.nvm/versions/node/v16.13.0/bin/npm
    Watchman: 2022.02.21.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    Android SDK:
      API Levels: 23, 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 31.0.0, 32.0.0
      System Images: android-29 | Google Play Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
      Android NDK: 22.1.7171670
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7621141
    Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: ^17.0.2 => 17.0.2 
    react-native: 0.66.1 => 0.66.1 
    react-native-macos: Not Found

**Comment by myles-grant** [comment]
I ran into this same issue. 

After upgrading react-native version 0.66.3 -> 0.68.2, I was able to successfully build the project.

**Comment by myles-grant** [comment]
Hi @charliecruzan-stripe, 

No. I had the deployment set to 13.0 with both versions. Although, when using react-native CLI to create a new project the deployment is initially set to iOS 11.0

**Comment by craigcoles** [comment]
I have also run into this issue.

All my deployment targets were updated from `11.0` to `12.0`. I was also running an older version (`13.0`) of Xcode so upgraded to the `13.2.1` as per the documentation, but still no joy.

Went through the process of upgrading react native from `0.66.0` to `0.68.2` as per @myles-grant's earlier [comment](https://github.com/stripe/stripe-react-native/issues/970#issuecomment-1146054097) and this seems to have resolved the issue 😅

**Comment by hiennguyen1311** [comment]
I'm facing the same issues, it worked well the first time but when I reinstalled packages and restarted my mac it showed this issue, my current react-native version is 0.68.2,  deployment target 12.0, Xcode 13.4.1. I tried removing and reinstalling or clear derived data but nothing worked.

**Comment by bill-pairaktaridis** [comment]
Yeah, we've had iOS 14 as a build target for a while and can't build with the latest version for this library. Trying to upgrade `react-native` but it should be listed as a requirement.

**Comment by charliecruzan-stripe** [comment]
You shouldn't need to upgrade your `react-native` version to 0.68 to resolve this issue as far as I can tell. I have a sample project with:

```
  "dependencies": {
    "@stripe/stripe-react-native": "^0.13.1",
    "react": "17.0.2",
    "react-native": "0.66.3"
  },
```

 that is building successfully on Xcode 13. I did originally get the "'maestro' is only available in iOS 12.0 or newer" error, but after totally clearing pods (removing both the ios/Pods/` and the `ios/Podfile.lock`), and cleaning the build folder & DerivedData, builds were successful. Here's the example project i used: https://github.com/charliecruzan-stripe/rn-0.66-with-stripe (it's just a react native init project + stripe)

**Comment by bill-pairaktaridis** [comment]
@charliecruzan-stripe unfortunately, no luck. Deleted `node_modules`, `Pods/`, `Podfile.lock`, `DerivedData` and still the same error.

**Comment by charliecruzan-stripe** [comment]
hm, that's odd.. did you try the project I linked?

**Comment by bill-pairaktaridis** [comment]
@charliecruzan-stripe yeah, it's not building for a different reason

```
fatal error: module map file '/Users/bla/Library/Developer/Xcode/DerivedData/AwesomeProject-gbvqgwgmoqpiibdjfruinashoehl/Build/Products/Debug-iphonesimulator/Stripe/Stripe.modulemap' not found
```

**Comment by charliecruzan-stripe** [comment]
@bill-pairaktaridis is it possible that one of the solutions [here](https://github.com/stripe/stripe-react-native/issues/253) might apply?

**Comment by bill-pairaktaridis** [comment]
@charliecruzan-stripe eventually, upgrade RN to `0.66.4` resolved the issue for us locally and on Codemagic.

**Comment by RajRohitYadav** [comment]
on M1 chip

We are using this script `__apply_Xcode_12_5_M1_post_install_workaround` as below

```
  post_install do |installer|
    react_native_post_install(installer)
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
```

this script has hard-coded `11.0` for `IPHONEOS_DEPLOYMENT_TARGET`. Hence explains why upgrading react-native solves the issue.

Changing it to `12.0` fixed it for us.

Path to the script
`node_modules/react-native/scripts/react_native_pods.rb`

**Comment by Tamimjabr** [comment]
just upgrade RN to 0.66.4

**Comment by Abhilash-R-Flipkart** [comment]
Facing this issue while installing maestro on ios
```
dyld: lazy symbol binding failed: can't resolve symbol _$sScCMa in /usr/local/Cellar/idb-companion/1.1.8/bin/../Frameworks/IDBGRPCSwift.framework/Versions/A/IDBGRPCSwift because dependent dylib @rpath/libswift_Concurrency.dylib could not be loaded
dyld: can't resolve symbol _$sScCMa in /usr/local/Cellar/idb-companion/1.1.8/bin/../Frameworks/IDBGRPCSwift.framework/Versions/A/IDBGRPCSwift because dependent dylib @rpath/libswift_Concurrency.dylib could not be loaded
fish: Job 1, 'idb_companion --udid 8C6D868E...' terminated by signal SIGABRT (Abort)
```

**Comment by fadikhan96** [comment]
config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0' in my case the here the target was 11 so i changed the target to 13.0

**Comment by diegomichell** [comment]
> 

This is the right answer, this is the reason that it fails, if you upgrade RN you get rid of this script hence you don't get the problem any more. If for some reason you can't upgrade changing IPHONEOS_DEPLOYMENT_TARGET to 13.0 resolves the problem
